### ISSUE TITLE
sara.py  line 219, undefined pd_tolf

### ISSUE THREAD

**Issue by y-mhiri** [ratt-ru/pfb-imaging#119]
if ntol < niters:
        pd_tolf = pd_tol[-1]
        pd_tol += [pd_tolf]*niters  # no harm in too many

When niters is equal to 1, for all ntol we have ntol >=niters. It results in pd_tolf not being defined raising an error later in the worker. A simple fix would be either changing the condition to ntol <=niters or setting pd_tolf before the condition. 

    pd_tolf = pd_tol[-1]
    if ntol < niters:
        pd_tol += [pd_tolf]*niters  # no harm in too many

**Comment by landmanbester** [comment]
Good catch, thanks for reporting. I will add a fix for this to https://github.com/ratt-ru/pfb-imaging/pull/117
